src/store.c

```diff
@@ -589,6 +590,26 @@ static int p11prov_store_set_ctx_params(void *pctx, const OSSL_PARAM params[])
     return RET_OSSL_OK;
 }
 
+void p11prov_store_quickload(void *provctx, const char *uri,
+                             OSSL_CALLBACK *object_cb, void *object_cbarg,
+                             OSSL_PASSPHRASE_CALLBACK *pw_cb, void *pw_cbarg)
+{
+    struct p11prov_store_ctx *ctx = NULL;
+    ctx = p11prov_store_open(provctx, uri);
+    if (!ctx) {
+        goto done;
+    }
+
+    while (
+        RET_OSSL_OK
+        == p11prov_store_load(ctx, object_cb, object_cbarg, pw_cb, pw_cbarg)) {
+        ;
+    }
+
```

Comment:
```
int ret = RET_OSSL_OK;
while (ret == RET_OSSL_OK) {
    ret = p11prov_store_load(ctx, object_cb, object_cbarg, pw_cb, pw_cbarg);
}
```

Also please return an error so you can bubble it up, there is no guarantee the load will succeed.

Reply:
I think it would be better to loop until p11prov_store_eof() is 1. Is there a clean way to accumulate the return values of store_load()?

Reply:
The only way, I think, is to create a linked list/array in the data structure you use to hold the callbacks.

Reply:
I'm returning a STACK_OF(int)* containing the return value of the store_load function calls. I cannot put the return value into the callback data structure because the store_load does not know the structure and the callback function is only called if the store was able to load something.

Reply:
Oh I misunderstood your question I guess.
So you were asking how to return the error from each call?
I do not think that is necessary, the reason to return the error, was to use
p11prov_set_error_mark(ctx->provctx); before the while, and after the while is over either call p11prov_pop_error_to_mark(ctx->provctx); if there was no error or p11prov_clear_last_error_mark(ctx->provctx); if an error happened.

Reply:
I'm glad that I misunderstood you and that I don't have to return all the return values as list. However, I'm still a bit confused about how the error handling should actually work.

Is there any reason for the decoder to do anything with the value returned by store_load?
If an error occurs the callback function should not be called (this is already part of the store's behaviour). But the decoders return value should be True(1) to allow OpenSSL to use different decoders.

Reply:
it is just to return debug logging to the caller which can then retrieve and paste it into their logs or discard as they wish.

Reply:
If `open` or `load` fail `p11prov_store_direct_fetch()` returns an error code and leaves the error stack intact.